runtime/src/main/java/org/corfudb/recovery/FastObjectLoader.java

```diff
@@ -473,11 +487,11 @@ private void updateCorfuObject(ILogData logData) {
      *
      */
     private void initializeHeadAndTails() {
-        if (logHead < 0) {
+        if (logHead == Address.NON_EXIST) {
```

Comment:
why do we even have this logic? shouldn't we always initialize the head/tail from the LUs ? Why are there setters on these variables. Can we clean this up?

Reply:
Setters seem to be used in the LayoutManagementView, to take advantage of the fact that the globalTail was requested already (I did change this to compute it within the FOL) and then retrieve the value which is further used for the sequencer reset. But it is also used to force tail and head in FOL Unit Tests. So this logic is aiming to bypass the check against log units.

Reply:
I removed the setters and declared these apart annotated as @VisibleForTesting